test/cli/run/run-function-scope.test.ts

```diff
@@ -0,0 +1,44 @@
+import { expect, test } from "bun:test";
+import { mkdirSync } from "fs";
+import { bunEnv, bunExe, tmpdirSync } from "harness";
+import { join } from "path";
+
+test("malformed function definition doesn't crash", async () => {
+  const dir = tmpdirSync();
+  mkdirSync(dir, { recursive: true });
+  await Bun.write(join(dir, "index1.js"), "function:");
+  await Bun.write(join(dir, "index1.ts"), "function:");
+  await Bun.write(join(dir, "index2.js"), "function a() {function:}");
+  await Bun.write(join(dir, "index2.ts"), "function a() {function:}");
```

Comment:
let's move these tests into `transpiler.test.js` and use `Bun.Transpiler` to parse them, probably in a new test scope.